### PR TITLE
refactor(pingcap/tidb): refactor two merged tests to prow sytle

### PR DESCRIPTION
**Pull request by purelind**
refactor some merged-pipelines to prow style

* merged_integration_copr_test
* merged_tiflash_test

### CODE REVIEW

**Review by wuhuizuo — CHANGES_REQUESTED** [review]

File: pipelines/pingcap/tidb/latest/merged_tiflash_test.groovy [hunk 1]
```diff
@@ -125,12 +109,12 @@ ENTRYPOINT ["/usr/local/bin/dumb-init", "/tidb-server"] \n' > Dockerfile
                         cp ../tidb/bin/tidb-server tidb-server
                         """
                         sh label: 'build tmp tidb image', script: """
-                        docker build -t hub.pingcap.net/qa/tidb:${GIT_BASE_BRANCH} -f Dockerfile .
+                        docker build -t hub.pingcap.net/qa/tidb:${REFS.base_ref} -f Dockerfile .
```
[the remark this answers is not in the document]
Comment (purelind) [reply pipelines/pingcap/tidb/latest/merged_tiflash_test.groovy:112]: here run tiflash test with docker-compose file, need build tidb image with current commit hash

[the remark this answers is not in the document]
Comment (lijie0123) [reply pipelines/pingcap/tidb/latest/merged_tiflash_test.groovy:112]: Is Dockerfile is in tiflash repo？ Will the docker image be pushed?

[the remark this answers is not in the document]
Comment (purelind) [reply pipelines/pingcap/tidb/latest/merged_tiflash_test.groovy]: 1. I have update here to use dockerfile from https://github.com/PingCAP-QE/artifacts/tree/main/dockerfiles.
2. tidb image that been built here not push to registry, only for current test.

**Review by wuhuizuo — APPROVED** [review]

**Pull request merged**